grid-client/deployer/deployment_deployer_test.go

```diff
@@ -660,5 +656,131 @@ func TestDeploymentDeployer(t *testing.T) {
+		Nodes:       []uint32{nodeID},
+		IPRange: gridtypes.NewIPNet(net.IPNet{
+			IP:   net.IPv4(10, 1, 0, 0),
+			Mask: net.CIDRMask(16, 32),
+		}),
+		AddWGAccess: false,
+	}
+
+	vm := workloads.VM{
+		Name:        "vm",
+		Flist:       "https://hub.grid.tf/tf-official-apps/base:latest.flist",
+		CPU:         2,
+		Planetary:   true,
+		Memory:      1024,
+		Entrypoint:  "/sbin/zinit init",
+		EnvVars:     map[string]string{"SSH_KEY": "<ssh key goes here>"},
+		NetworkName: n.Name,
+	}
+
+	err = tfPluginClient.NetworkDeployer.Deploy(context.Background(), &n)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	dl := workloads.NewDeployment("vm", nodeID, "", nil, n.Name, nil, nil, []workloads.VM{vm}, nil)
+	err = tfPluginClient.DeploymentDeployer.Deploy(context.Background(), &dl)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	fmt.Println("deployment is done successfully")
+}
 
+func ExampleDeploymentDeployer_BatchDeploy() {
+	const mnemonic = "<mnemonics goes here>"
+	const network = "<dev, test, qa, main>"
+
+	tfPluginClient, err := NewTFPluginClient(mnemonic, "sr25519", network, "", "", "", 0, false)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	n := workloads.ZNet{
+		Name:        "network",
+		Description: "network for testing",
+		Nodes:       []uint32{nodeID},
+		IPRange: gridtypes.NewIPNet(net.IPNet{
+			IP:   net.IPv4(10, 1, 0, 0),
+			Mask: net.CIDRMask(16, 32),
+		}),
+		AddWGAccess: false,
+	}
+
+	vm1 := workloads.VM{
+		Name:        "vm1",
+		Flist:       "https://hub.grid.tf/tf-official-apps/base:latest.flist",
+		CPU:         2,
+		Planetary:   true,
+		Memory:      1024,
+		Entrypoint:  "/sbin/zinit init",
+		EnvVars:     map[string]string{"SSH_KEY": "<ssh key goes here>"},
+		NetworkName: n.Name,
+	}
+	vm2 := workloads.VM{
+		Name:        "vm2",
+		Flist:       "https://hub.grid.tf/tf-official-apps/base:latest.flist",
+		CPU:         2,
+		Planetary:   true,
+		Memory:      1024,
+		Entrypoint:  "/sbin/zinit init",
+		EnvVars:     map[string]string{"SSH_KEY": "<ssh key goes here>"},
+		NetworkName: n.Name,
+	}
+
+	err = tfPluginClient.NetworkDeployer.Deploy(context.Background(), &n)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	d1 := workloads.NewDeployment("vm1", nodeID, "", nil, n.Name, nil, nil, []workloads.VM{vm1}, nil)
+	d2 := workloads.NewDeployment("vm2", nodeID, "", nil, n.Name, nil, nil, []workloads.VM{vm2}, nil)
+	err = tfPluginClient.DeploymentDeployer.BatchDeploy(context.Background(), []*workloads.Deployment{&d1, &d2})
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	fmt.Println("batch deployment is done successfully")
+}
+
+func ExampleDeploymentDeployer_Cancel() {
+	const mnemonic = "<mnemonics goes here>"
+	const network = "<dev, test, qa, main>"
+
+	tfPluginClient, err := NewTFPluginClient(mnemonic, "sr25519", network, "", "", "", 0, false)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	// dl should be a valid and existing deployment
+	dl := workloads.Deployment{}
```

Comment:
What about using the state to get the deployment?
Check other examples

Reply:
The state holds regular contracts ids not a deployment, so I think we can't use `DeploymentDeployer.Cancel()` with them

Reply:
There are getters for the deployments in the state